### PR TITLE
Introduce testing decorators

### PR DESCRIPTION
**Pull request by LukeWood**
These decorators allow you to skip tests if they require lots of resources, time, or custom C++ ops.

They're pretty trivial, and just make tests opt-in.

### CODE REVIEW

File: keras_cv/testing/utils.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+
+
+def integration(test_fn):
```
Comment (ianstenbit) [inline keras_cv/testing/utils.py:17]: Do we have a use for the integration annotation? If not, maybe we should make this PR just add requires_custom_ops for now?

Also, FWIW, it seems like we could make a meta function here which creates a decorator with a given tag (e.g. `requires_custom_ops = build_opt_in_test_decorator("TEST_CUSTOM_OPS")`)

**Comment by ianstenbit** [comment]
@LukeWood do you still want to move forward with this concept?

**Comment by jbischof** [comment]
KerasNLP already has this functionality set up. We may want to follow that pattern: https://github.com/keras-team/keras-nlp/blob/master/keras_nlp/conftest.py

**Comment by LukeWood** [comment]
> KerasNLP already has this functionality set up. We may want to follow that pattern: https://github.com/keras-team/keras-nlp/blob/master/keras_nlp/conftest.py

yeah this looks like a better setup than what we have with decorators.

**Comment by LukeWood** [comment]
closed as obselete

**Pull request closed without merging**